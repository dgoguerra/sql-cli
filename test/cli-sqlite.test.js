const { getTestKnex } = require("./utils");
const { cliTestSuite } = require("./cli-test-suite");

cliTestSuite("sqlite3", () => getTestKnex());
