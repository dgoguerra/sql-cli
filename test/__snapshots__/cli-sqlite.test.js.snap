// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI test: sqlite3 can create conn aliases 1`] = `
"alias  conn                                     
alias1 mysql://app:secret@127.0.0.1:1234/dbname 
alias2 sqlite3:///path/to/file.db               
"
`;

exports[`CLI test: sqlite3 can delete conn aliases 1`] = `
"alias  conn                       
alias2 sqlite3:///path/to/file.db 
"
`;

exports[`CLI test: sqlite3 can diff schemas 1`] = `
"No tables with changes
"
`;

exports[`CLI test: sqlite3 can diff tables 1`] = `
"column  type                     
field_1 integer nullable → float 

4x similar, 1x changed
"
`;

exports[`CLI test: sqlite3 can diff tables data 1`] = `
"Diff of tables content (first 20 rows):

id field_1    updated_at                                
1  12 → 12.3  [null] → 2020-07-24 18:34:00              
2  30 → 30.45 2020-07-24 18:34:00 → 2020-07-24 19:25:00 
"
`;

exports[`CLI test: sqlite3 can list tables 1`] = `
"table   rows bytes  
table_1 2    4.1 kB 
table_2 2    4.1 kB 
table_3 0    4.1 kB 

(12.3 kB in 3 tables)
"
`;

exports[`CLI test: sqlite3 can load database dump 1`] = `
"column     type     nullable 
id         integer  false    
field_1    integer  true     
field_2    text     true     
created_at datetime true     
updated_at datetime true     
"
`;

exports[`CLI test: sqlite3 can load database dump 2`] = `
"column     type     nullable 
id         integer  false    
field_1    float    false    
field_2    text     true     
created_at datetime true     
updated_at datetime true     
"
`;

exports[`CLI test: sqlite3 can load database dump 3`] = `
"column     type     nullable 
idField    integer  false    
field_1    bigint   true     
created_at datetime true     
updated_at datetime true     
"
`;

exports[`CLI test: sqlite3 can show table 1`] = `
"column     type     nullable 
id         integer  false    
field_1    float    false    
field_2    text     true     
created_at datetime true     
updated_at datetime true     
"
`;

exports[`CLI test: sqlite3 can use a created alias 1`] = `
"table   rows bytes  
table_1 2    4.1 kB 
table_2 2    4.1 kB 
table_3 0    4.1 kB 

(12.3 kB in 3 tables)
"
`;

exports[`CLI test: sqlite3 database dump contents are valid 1`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_1\\", (t) => {
    t.increments(\\"id\\");
    t.integer(\\"field_1\\");
    t.text(\\"field_2\\");
    t.timestamp(\\"created_at\\");
    t.timestamp(\\"updated_at\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_1\\");
};
"
`;

exports[`CLI test: sqlite3 database dump contents are valid 2`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_2\\", (t) => {
    t.increments(\\"id\\");
    t.float(\\"field_1\\").notNullable();
    t.text(\\"field_2\\").defaultTo(\\"default text\\");
    t.timestamp(\\"created_at\\");
    t.timestamp(\\"updated_at\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_2\\");
};
"
`;

exports[`CLI test: sqlite3 database dump contents are valid 3`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_3\\", (t) => {
    t.increments(\\"idField\\");
    t.bigInteger(\\"field_1\\");
    t.timestamp(\\"created_at\\");
    t.timestamp(\\"updated_at\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_3\\");
};
"
`;

exports[`CLI test: sqlite3 database dump contents are valid 4`] = `
"{\\"id\\":1,\\"field_1\\":12,\\"field_2\\":\\"foo\\",\\"created_at\\":\\"2020-07-24 18:34:00\\"}
{\\"id\\":2,\\"field_1\\":30,\\"field_2\\":\\"bar\\",\\"created_at\\":\\"2020-07-24 18:34:00\\",\\"updated_at\\":\\"2020-07-24 18:34:00\\"}
"
`;

exports[`CLI test: sqlite3 database dump contents are valid 5`] = `
"{\\"id\\":1,\\"field_1\\":12.3,\\"field_2\\":\\"foo\\",\\"created_at\\":\\"2020-07-24 18:34:00\\",\\"updated_at\\":\\"2020-07-24 18:34:00\\"}
{\\"id\\":2,\\"field_1\\":30.45,\\"field_2\\":\\"bar\\",\\"created_at\\":\\"2020-07-24 18:34:00\\",\\"updated_at\\":\\"2020-07-24 19:25:00\\"}
"
`;
